### PR TITLE
add tutorial

### PR DESCRIPTION
**Pull request by charlierudolph**

### CODE REVIEW

File: docs/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+# Object with dynamic keys and values that follow a schema
+
+This can be achieved with the the second argument to schema. We call it `map`.
+
+```js
+// store/user_session_slice.js
+import {createSlice} from 'nectarine'
+
+const userSessionSlice = createSlice({
+  schema: (_) => {
```
Comment (a-morales) [inline docs/tutorial.md]: Is there a way to specify a single top most `map` like
```js
const projectsSlice = createSlice({
  schema: (_, map) => map({ summary: _, title: _)}
})
...

const store  = create store({projects: projectsSlice});

store.projects.$key("1").set({summary: "foo", title: "bar"});
```

File: docs/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+      email: _,
+      name: _
+    }
+    projects: map({
+      summary: _,
+      title: _
+    })
+  },
+  actions: {
+    addProject: ({summary, title}) => {
```
Comment (a-morales) [inline docs/tutorial.md]: The `=>` won't have the correct `this` binded. this has to be
```js
addProject: function({summary, title}) { ...  }
```

File: docs/tutorial.md (hunk 1 above)
Comment (charlierudolph) [inline docs/tutorial.md]: Your example already works. We even have a [test](https://github.com/Originate/nectarine/blob/master/src/create-store.spec.ls#L72) for it

File: docs/tutorial.md (hunk 2 above)
Comment (charlierudolph) [inline docs/tutorial.md]: Indeed. Thanks

**Pull request closed without merging**